examples/llm_stack/run.py

```diff
@@ -0,0 +1,43 @@
+import data_module
+import embedding_module
+import pinecone_module  # noqa: F401
+import weaviate_module
+
+from hamilton import base, driver
+
+
+def main():
+    weaviate_config = {"url": "http://localhost:8080/"}
+    # pinecone_config = {
+    #     "environment": "PINECONE_ENVIRONMENT",
+    #     "api_key": "PINECONE_API_KEY",
+    # }
+
+    config = dict(
+        vector_db_config=weaviate_config,
+        embedding_service="(openai,cohere,sentence_transformer)",  # this triggers config.when() in embedding_module
+        api_key="EMBEDDING_SERVICE_API_KEY",  # needed if using OpenAI or Cohere
+        model_name="text-embedding-ada-002",
+    )
+
+    dr = driver.Driver(
+        config,
+        data_module,
+        weaviate_module,  # this can be swapped for `pinecone_module`
+        embedding_module,
+        adapter=base.SimplePythonGraphAdapter(base.DictResult()),
+    )
+
+    # The `final_vars` requested are functions with side-effects
+    dr.execute(
+        final_vars=["initialize_vector_db_indices", "push_to_vector_db"],
+        inputs=dict(
+            class_name="SQuADEntry",
+        ),
+    )
+
+    # uncomment below to produce visualization
+    # dr.display_all_functions("dag", {"format": "png"})
+
+if __name__ == "__main__":
+    main()
```

Comment:
Something like this is probably more appropriate, though perhaps it should parse arguments from the commandline... Also the README should have instructions on how to run this code, e.g. whether to add API Keys etc.

```suggestion
def main(vector_db_type: str, embedding_service: str, embedding_service_api_key: str = None):
    """Main function to run our dataflow with Hamilton.
    :param vector_db_type: weaviate or pinecone
    :param embedding_service: openai, cohere, sentence_transformer
    :param embedding_service_api_key: needed if using OpenAI or Cohere
    """
    if vector_db_type == "weaviate":
        vector_db_config = {"url": "http://localhost:8080/"}
        vector_db_module = weaviate_module
    elif vector_db_type == "pinecode":
        vector_db_config = {
             "environment": "PINECONE_ENVIRONMENT",
             "api_key": "PINECONE_API_KEY",
        }
        vector_db_module = pinecone_module

    config = dict(
        vector_db_config=vector_db_config,
        embedding_service=embedding_service,  # "(openai,cohere,sentence_transformer)"
                                              # this triggers config.when() in embedding_module
        api_key=embedding_service_api_key,  # needed if using OpenAI or Cohere
        model_name="text-embedding-ada-002",  # needed if using OpenAI
    )

    dr = driver.Driver(
        config,
        data_module,
        vector_db_module,  # this can be swapped for `pinecone_module`
        embedding_module,
        adapter=base.SimplePythonGraphAdapter(base.DictResult()),
    )

    # The `final_vars` requested are functions with side-effects
    dr.execute(
        final_vars=["initialize_vector_db_indices", "push_to_vector_db"],
        inputs=dict(
            class_name="SQuADEntry",
        ),
    )

    # uncomment below to produce visualization
    # dr.display_all_functions("dag", {"format": "png"})

if __name__ == "__main__":
    import os
    api_key = os.environ.get("OPENAI_API_KEY")
    main("weaviate", "openai", api_key)
    # main("pinecone", "openai", "EMBEDDING_SERVICE_API_KEY")
```

Reply:
Added!